sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -2408,7 +2408,8 @@ public void testExactCountDistinctWithFilter()
                                 new CountAggregatorFactory("_a0"),
                                 and(
                                     not(selector("d0", null, null)),
-                                    selector("a1", "0", null)
+                                    selector("a1", "0", null),
+                                    expressionFilter("\"d1\"")
```

Comment:
hmm, what is the story on this expression filter?

Reply:
still curious

Reply:
There has been a virtual column here 
```
expressionVirtualColumn(
                                                "v0",
                                                NullHandling.replaceWithDefault()
                                                ? "(\"cnt\" == 1)"
                                                : "((\"cnt\" == 1) > 0)",
                                                ColumnType.LONG
                                            )
```
which is being represented in planning by d1 and when there is an aggregation on the `cnt` column it checks if that value is null or not. If you think this is a bug, let me know